### PR TITLE
SplitCheckout: as guest user, after login should redirect to the first step of checkout

### LINKED ISSUE #8894: [Split checkout] Logging in during guest checkout redirects to homepage
## Description

Logging in during guest checkout redirects to homepage, forcing the user to click the cart and proceed to checkout again.

## Expected Behavior

If the user is prompted to log in during checkout it should be redirected to the `/details` page.

## Actual Behaviour

If the user is prompted to log in during checkout it currently is redirected to the homepage.

## Steps to Reproduce

1. Try to checkout on a guest session on a shop set to require log in.
2. After logging in, notice the redirection to the homepage.

## Animated Gif/Screenshot

Legacy checkout:

Peek 2022-02-15 17-37

Current Split-checkout:

Peek 2022-02-15 17-41

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

Closes #8894 

#### What should we test?

- As guest user, proceed to checkout
- You should see the "Ok, ready to checkout?" page (with at least the login button)
- Click on login, should open the modal, fill the form and then proceed to login
- You should be redirected to the first step of the checkout and logged-in

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -62,7 +62,9 @@ def redirect_to(options = {}, response_status = {})
   end
 
   def set_checkout_redirect
-    return unless URI(request.referer.to_s).path == main_app.checkout_path
+    referer_path = URI(request.referer.to_s).path
+    return unless referer_path == main_app.checkout_path ||
+                  referer_path == main_app.checkout_step_path(:details)
 
     session["spree_user_return_to"] = main_app.checkout_path
   end
```
Comment (mkllnk) [inline app/controllers/application_controller.rb:70]: I think that this method shouldn't be in the ApplicationController but I haven't looked through all the code to suggest a better way.

**Review by mkllnk — APPROVED** [review]

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (jibees) [reply app/controllers/application_controller.rb:70]: I only changed the method behavior, but I did not move it. 
Why do you suggest this one should not be here? I can see there also there are some others methods in `application_controller` that refers to _redirect_, like `redirect_to` or `after_sign_in_path_for`. 
Just to be sure to understand the reasons: then I can start looking for a better place!

Comment (mkllnk) [reply app/controllers/application_controller.rb:70]: Sorry, I wasn't criticising you. I know that you didn't move it there. This method seems to be checkout related while the application controller contains common code used by most other controllers. A generic redirect is very common in most controllers. But `set_checkout_redirect` is only used when logging in. Only two controllers use it.

I would probably leave this until the split checkout is complete and active. Then we can have some new logic that either returns the user to where they came from (referer) or looks at the order state to go to the right shopping/checkout step. But doing that with a feature toggle sounds too complicated.

Comment (jibees) [reply app/controllers/application_controller.rb:70]: > Sorry, I wasn't criticising you.

I wasn't complaining, sorry @mkllnk ! That was just to be sure to understand.

> I would probably leave this until the split checkout is complete and active.

I agree. Let's resolve this conversation then?

**Review by Matt-Yorkley — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @jibees ,

fixed!

Peek 2022-03-07 09-07

**Pull request merged**